src/Form/AuthenticationForm.php

```diff
@@ -190,26 +203,31 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#tree' => TRUE,
     ];
 
+    // Validate private file path is set.
+    if (!PrivateStream::basePath()) {
+      $this->disableForm($form, $this->t('The Drupal private file setting has not been configured.'));
+      return $form;
+    }
+
+    // Validate private file path is a directory and is writable.
+    $file_private_path = Settings::get('file_private_path');
```

Comment:
Core does perform an `$is_writable `check, but it also tries to create directories first, so maybe it only does it because of that.
https://github.com/drupal/core/blob/8bd590f7aec78b8d273187ec2d1adab262e2e1d7/modules/system/system.install#L670

Reply:
This was the way the original code was written, but @llynch00 and I both saw a few issues with that approach: 1) The error message was vague and would say to configure the private file path even when it was set in settings.php and 2) still came back null after making a change to the settings.php file and/or creating the directory.  Seemed to need a cache refresh for the error to go away.  #1 problem was also why we are checking if the path is writable. 

I updated to use `PrivateStream::basePath()` just like `system_install()` does.

Reply:
> still came back null after making a change to the settings.php file and/or creating the directory. Seemed to need a cache refresh for the error to go away.
Yes, you should always do a cache clear if you modify its content. So this is part of the workflow.